packages/studio-ui/src/hooks/useInjectUserStyles.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect } from "react";
+import useStudioStore from "../store/useStudioStore";
+
+export default function useInjectUserStyles(
+  iframeDocument: Document | undefined
+) {
+  const [
+    componentTree,
+    getComponentMetadata,
+    activePage,
+    UUIDToFileMetadata,
+    pageCss,
+  ] = useStudioStore((store) => [
+    store.actions.getComponentTree(),
+    store.actions.getComponentMetadata,
+    store.pages.activePageName,
+    store.fileMetadatas.UUIDToFileMetadata,
+    store.pages.getActivePageState()?.cssImports,
```

Comment:
out of curiosity, if the active page changes, do we see multiple extra re-renders because we're subscribing to different parts of the active page state separately? or would it be the same as if we subscribed to the whole active page state at once?

Reply:
I removed the activePageName, so we shouldn't run into issues now!

Reply:
oh I meant that we get the active page's component tree and css imports separately

Reply:
Ah, I see. I tested it out and based on what I could tell, the iframe renders the same amount (twice) either way.  I changed it to just import the active page state once, since that seems a little cleaner!

Reply:
If the number of renders is the same, I'd actually prefer keeping them as separate selectors. Because, while it is a little cleaner to group them, we'd get extra re-renders if some part of the PagesJS state for the active page was changed even though that doesn't affect the styling

Reply:
ahhh that makes sense. I'll change it